tests/Microsoft.Identity.Test.iOS.UIAutomation/XamariniOSTests.cs

```diff
@@ -0,0 +1,229 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.Identity.Test.LabInfrastructure;
+using NUnit.Framework;
+using Microsoft.Identity.Test.UIAutomation.Infrastructure;
+using Xamarin.UITest;
+using Xamarin.UITest.Queries;
+using System;
+using System.Linq;
+
+//NOTICE! Inorder to run UI automation tests for xamarin locally, you may need to upgrade nunit to 3.0 and above for this project and the core ui Automation project.
+//It is set to 2.6.4 because that is the maximum version that appcenter can support.
+//There is an error in visual studio that can prevent the NUnit test framework from loading the test dll properly.
+//Remember to return the version back to 2.6.4 before commiting to prevent appcenter from failing
+
+namespace Test.Microsoft.Identity.Msal.iOS.UIAutomation
+{
+    /// <summary>
+    /// Configures environment for core/iOS tests to run
+    /// </summary>
+    [TestFixture(Platform.iOS)]
+    public class XamarinIOSMsalTests
```

Comment:
nit: just an idea, but could we remove "Xamarin" from the name? Same w/the android tests and just have `iOSTests` and `AndroidTests` as we know 1. they are xamarin and 2. they are msal. just a thought. makes for less reading.

Reply:
Same with `IOSXamarinUITestController` maybe just have `iOSUiTestController`

Reply:
Also, if we do get to move to iOS12 w/app center, there is a dialog box which appears, so we'll need to add in an extra step at the start of the auth cycle.